src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -536,35 +537,91 @@ public static decimal Max(decimal val1, decimal val2)
             return decimal.Max(val1, val2);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
What is the size of code that this expands to? This does not look like a good AggressiveInlining candidate...

Reply:
For modern processors with AVX support (essentially any CPU from 2011 onwards), this expands to (plus or minus a couple bytes depending on registers used):

Math.Min - 32 bytes:
```asm
	vminss      xmm2, xmm1, xmm0
	vcmpeqss    xmm3, xmm1, xmm0
	vorps       xmm4, xmm1, xmm0
	vandps      xmm4, xmm4, xmm3
	vorps       xmm2, xmm2, xmm4
	vcmpunordss xmm0, xmm0, xmm0
	vblendvps   xmm0, xmm2, xmm1, xmm0
```

Math.Max - 30 bytes:
```asm
	vmaxss      xmm2, xmm1, xmm0
	vcmpeqss    xmm3, xmm1, xmm0
	vandps      xmm4, xmm1, xmm0
	vblendvps   xmm2, xmm2, xmm4, xmm3
	vcmpunordss xmm0, xmm0, xmm0
	vblendvps   xmm0, xmm2, xmm1, xmm0
```

Reply:
For older processors, without AVX support but with SSE4.1 support (~2007 onwards), this expands to:

Math.Min - 52 bytes:
```asm
	movaps      xmm2, xmm1
	minss       xmm2, xmm0
	movaps      xmm3, xmm1
	cmpeqss     xmm3, xmm0
	movaps      xmm4, xmm1
	orps        xmm4, xmm0
	andps       xmm4, xmm3
	orps        xmm2, xmm4
	movaps      xmm3, xmm0
	cmpunordss  xmm3, xmm0
	movaps      xmm0, xmm3
	blendvps    xmm2, xmm1, xmm0
	movaps      xmm0, xmm2
```

Math.Max - 56 bytes:
```asm
	movaps      xmm2, xmm1
	maxss       xmm2, xmm0
	movaps      xmm3, xmm1
	cmpeqss     xmm3, xmm0
	movaps      xmmword ptr [rsp],xmm0
	movaps      xmm4, xmm1
	andps       xmm4, xmm0
	movaps      xmm0, xmm3
	blendvps    xmm2, xmm4, xmm0
	movaps      xmm0, xmmword ptr [rsp]
	movaps      xmm3, xmm0
	cmpunordss  xmm3, xmm0
	movaps      xmm0, xmm3
	blendvps    xmm2, xmm1, xmm0
	movaps      xmm0, xmm2
```

Reply:
Even older x86 CPUs (anything else the runtime supports) expands to 56-bytes (`Math.Min`) and 65-bytes (`Math.Max`).

Reply:
The cost of not-inlining looks to be ~29 bytes of code per invocation (must save and restore the floating-point values):
```asm
movss       dword ptr [rsp+0x40], xmm0  
movss       dword ptr [rsp+0x48], xmm1  
call        <method>
vmovss  xmm0,dword ptr [rsp+0x48]
vmovss  xmm1,dword ptr [rsp+0x40]
```

Given that this is, on modern CPUs, basically the same cost as the method, and on older CPUs nearly half the cost of the actual method; it should be almost always profitable to inline (and various micro-benchmarks show the same).